.travis.yml

```diff
@@ -0,0 +1,144 @@
+- test $RES -eq 0
+- kill $RPCBPID
+
+#
+# Test #4, in this test the first policy rule restricts all traffic to a
+# destination and path.  The following rules allow only messages to two
+# specific members that are now not being blocked.
+#
+- ./rpc-broker -r 5555 &
+- RPCBPID=$!
+- dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect; RES=$?
+- test $RES -eq 0
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect 2>first-fail.txt || true 
+- diff -u first-fail.txt test-policies/blocked-interface-reply.txt ; RES=$? 
+- test $RES -eq 0
+- kill $RPCBPID
+
+- sudo cp test-policies/test_3.rules /etc/rpc-broker.rules
+- ./rpc-broker -r 5555 &
+- RPCBPID=$!
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect; RES=$?
+- test $RES -eq 0
+- kill $RPCBPID
+
+#
+# Test #5, this test demonstrates a policy which blocks with fine granularity,
+# down to disallowing only a certain method of a given interface.  Setup to
+# show all calls are initially blocked and then after the policy is put in 
+# place, only calls to a specific method are blocked.
+#
+- ./rpc-broker -r 5555 &
+- RPCBPID=$!
+- dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect; RES=$?
+- test $RES -eq 0
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect 2>first-fail.txt || true 
+- diff -u first-fail.txt test-policies/blocked-interface-reply.txt ; RES=$? 
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.ListNames 2>second-fail.txt || true 
+- diff -u second-fail.txt test-policies/blocked-interface-reply.txt ; RES=$? 
+- test $RES -eq 0
+- kill $RPCBPID
+
+- sudo cp test-policies/test_4.rules /etc/rpc-broker.rules
+- ./rpc-broker -r 5555 &
+- RPCBPID=$!
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.Introspectable.Introspect; RES=$?
+- test $RES -eq 0
+- DBUS_SYSTEM_BUS_ADDRESS=tcp:host=127.0.0.1,port=5555 dbus-send --system --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.ListNames 2>second-still-fail.txt || true 
+- diff -u test-policies/blocked-interface-reply.txt second-still-fails.txt ; RES=$?
+- test $RES -eq 0
+- kill $RPCBPID
+
+#
```

Comment:
It would be nice to have tests for "allow all except ..." and "deny all except ..."